src/plugins/data/public/search/aggs/types.ts

```diff
@@ -55,6 +66,11 @@ export interface SearchAggsStart {
   types: AggTypesRegistryStart;
 }
 
+/** @internal */
+export interface BaseAggParams {
+  json?: string;
```

Comment:
Should `customLabel` be here too? AFAICT [it should be required for all aggs](https://github.com/elastic/kibana/blob/da334ae6f838b23bd2cb44eed92a2f7fa464aed9/src/plugins/data/public/search/aggs/agg_type.ts#L244-L264), except for `filters` in which it appears to be explicitly set to `false`